src/managed/Microsoft.NET.HostModel/AppHost/HostWriter.cs

```diff
@@ -43,6 +45,16 @@ public static class HostWriter
             }
 
             BinaryUtils.CopyFile(appHostSourceFilePath, appHostDestinationFilePath);
+
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                var filePermissionOctal = Convert.ToInt32("755", 8); // -rwxr-xr-x
+                if (chmod(appHostDestinationFilePath, filePermissionOctal) == -1)
+                {
+                    throw new Win32Exception(Marshal.GetLastWin32Error(), $"Could not set file permission {filePermissionOctal} for {appHostDestinationFilePath}.");
+                }
```

Comment:
This code should be wrapped within a `RetryUtil.RetryOnWin32Error`

Reply:
See @tmds' comment at https://github.com/dotnet/core-setup/pull/8510#discussion_r333463143, chmod does not throw an exception, we are throwing this exception to get useful error message.

Reply:
`chmod` doesn't block if a file is being used; but it could still fail because of EINTR. We could have a RetryUtil.RetryOnSignal() for this method.

For now, please add a comment that says `chmod()` doesn't block on IO.

Reply:
With some digging, I have found that the retry logic for chmod and friends is typically based on EINTR && the return code, in C/C++ (liblp etc.). In addition to that, I have added retry counter. It can be moved to `RetryUtil` class as `RetryOnSignal` or `RetryOnErrno`, but that would probably make it too specific to chmod.

I think we can postpone moving it to RetryUtil, until the next interruptible api of similar category is p/invoked and used in HostModel.